qiskit/providers/ibmq/runtime/ibm_runtime_service.py

```diff
@@ -540,6 +567,28 @@ def _decode_job(self, raw_data: Dict) -> RuntimeJob:
                           params=decoded,
                           creation_date=raw_data.get('created', None))
 
+    def _encode_program_data(self, data: Union[bytes, str]) -> bytes:
+        """Encodes a runtime program data into Base64. Program data can be
+        either a filepath (str) or raw bytes.
+
+        Args:
+            data: the program data
+
+        Raises:
+            FileNotFoundError: If the data filepath does not exist.
+
+        Returns:
+            the encoded data
+        """
+        # Convert any file data into bytes
+        if isinstance(data, bytes):
+            return data
+        with open(data, 'r') as file:
+            data = file.read().encode()
+        # Convert the bytes data into Base64
+        data = base64.b64encode(data)
+        return data
```

Comment:
Can this be like below? That way the input provided by user can either be string or bytes (with no encoding) and then we can convert to base64.

```
if isinstance(data, str):
    with open(data, 'r') as file:
        data = file.read()
data = base64.b64encode(data)
return data
```

Reply:
Commented [above](https://github.com/Qiskit/qiskit-ibmq-provider/pull/1008/files/664e13a982b4a722a58904d1dd9cd3b8067d7a5a#r677708225)!